lib/services/elasticsearch.js

```diff
@@ -1411,32 +1452,34 @@ class ElasticSearch extends Service {
       }
     }
 
-    return this.mGet(index, collection, toDelete)
-      .then(({ hits }) => {
+    // @todo should be done in the clientAdapter,
+    // because we only need it for realtime
+    return this.mGet(index, collection, toGet)
+      .then(({ items }) => {
         let idx = 0;
 
         for (let i = 0; i < ids.length; i++) {
           const
             id = ids[i],
-            hit = hits[idx];
+            item = items[idx];
 
-          if (hit && hit._id === id) {
-            toDelete.push(id);
+          if (item && item._id === id) {
+            query.ids.values.push(id);
             idx++;
           } else {
             partialErrors.push({ id, reason: 'cannot find document' });
           }
         }
 
-        esRequest.body.ids.values = toDelete;
-
-        return this.client.deleteByQuery(esRequest);
+        // @todo duplicated query to get documents body, mGet here and search in
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)